geonode/geoserver/manager.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+#########################################################################
+#
+# Copyright (C) 2021 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+import logging
+import tempfile
+
+from django.db.models.query import QuerySet
+
+from geonode.layers.models import Layer
+from geonode.base.models import ResourceBase
+from geonode.services.enumerations import CASCADED
+from geonode.resource.manager import ResourceManagerInterface
+
+from .tasks import (
+    geoserver_set_style,
+    geoserver_create_style,
+    geoserver_cascading_delete,
+    geoserver_create_thumbnail)
+from .signals import geoserver_post_save_local
+from .helpers import (
+    gs_catalog,
+    set_time_info,
+    ogc_server_settings)
+
+logger = logging.getLogger(__name__)
+
+
+class GeoServerResourceManager(ResourceManagerInterface):
+
+    def search(self, filter: dict, /, type: object = None) -> QuerySet:
+        return type.objects.none()
+
+    def exists(self, uuid: str, /, instance: ResourceBase = None) -> bool:
+        if uuid and instance:
+            if hasattr(instance.get_real_instance(), 'storeType') and instance.get_real_instance().storeType not in ['tileStore', 'remoteStore']:
```

Comment:
maybe something like this in order to have less to read?
```python
real_instance = instance.get_real_instance()
if hasattr(real_instance, 'storeType') and real_instance.storeType not in ['tileStore', 'remoteStore']:
....
```

Reply:
better looking at the first condition, I guess there's an error here. I should check for uuid only and eventually fetch the instance.

Reply:
This one should replace only the line 51 once the first if is validated